temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -430,7 +428,11 @@ private void handleSingleEvent(HistoryEvent event, boolean hasNextEvent) {
       replaying = false;
     }
 
-    Long initialCommandEventId = getInitialCommandEventId(event);
+    final long initialCommandEventId = getInitialCommandEventId(event);
```

Comment:
Can we document `getInitialCommandEventId` and the fact that if it return < 0 the event can be ignored?

Reply:
Added a docstring, but also switched from `long` / `-1` to `OptionalLong` / `empty()` because it's less cognitive load.